### DISCUSSION TITLE
Question about the best practices for making changes and testing.

### DISCUSSION THREAD

**Discussion by BradNagel** [Koenkk/zigbee2mqtt#12680] [General]
I recently have been moving off of my Insteon to Zigbee devices after Insteons recently closure. I'm started to learn my way around how Zigbee works. After buying a the Leviton DG6HD-1BW dimmer I was a bit disappointed that I could not adjust some of the settings that I had in my old Insteon dimmer. After digging around in the dev tools I found all the settings I was looking for. Power On Transition, Power Off Transition, Power On Behavior and Min and Max dimmer levels. From what I can see changes for the most part are easy to make. One thing I wasn't sure on was the best way to test. Can I access the Zigbee2mqtt addon to make changes? or is there a better way?

https://www.zigbee2mqtt.io/devices/DG6HD-1BW.html

**Comment by nebhead** [comment]
@BradNagel 

I'm also interested in this, as I also just bought the DG6HD-1BW (well actually two of them).  And after reaching out to their support regarding setting up these other features (Power On Transition, Power Off Transition, Power On Behavior and Min and Max dimmer levels), they responded with this:

> 
> The My Leviton app options that you are referring to below are only available for Decora Smart Wi-Fi dimmers; as the My Leviton app only is only compatible with Decora Smart Wi-Fi devices.
>  
> The DG6HD dimmer does support those options, but they cannot be configured using the dimmer paddle. They can only be configured wirelessly (via Zigbee), by a Zigbee coordinator that supports the ability to configure those options.
>  

So, I'm curious if you did discover how to make these changes on the switch configuration, and how I could do the same.  I also second the idea of having these options built into Zigbee2MQTT.  

Thanks!

**Reply by timborino** [reply]
[Min/Max dimmer levels](https://hubconnecteddecorasmart.zendesk.com/hc/en-us/articles/4407972329371-Setting-the-Minimum-and-Maximum-Dim-Levels-of-a-Decora-Smart-Zigbee-Dimmer) can be programmed via the paddle according to their knowledge base. The [LED behavior](https://hubconnecteddecorasmart.zendesk.com/hc/en-us/articles/4407972271003-Configuring-the-Status-Light-on-a-Decora-Smart-Zigbee-Dimmer-or-Switch) can also be programmed via the paddle, apparently. It would be great if they included these in their user manual.

**Reply by nebhead** [reply]
The user manual for my switch did have these options listed as programmable through the paddles.  However it didn't have other features like Power On Transition, Power On Behavior, etc.  These are features that can be programmed via the App on WiFi models.  But since this Zigbee version doesn't have an associated App, you have to rely upon the hub software.  When I reached out to Leviton support, this is what they repeated to me.   I'm super happy that Z2M supports manually modifying these attributes via the dev console, as it means I can tune these switches how I like them.  It would be even better if these options were just there in the software, in the exposes section.

**Reply by timborino** [reply]
I agree. I bought 20+ units of these dimmers, and the transition settings are exposed in ZHA on Home Assistant. Migrating them to Z2M has removed my ability to change the settings.

I'm making progress on teaching myself how to expose the capabilities in Z2M. It's a work in progress, but I've attached a custom converter file for this dimmer if you'd like to try it. [DG6HD-1BW.js.txt](https://github.com/Koenkk/zigbee2mqtt/files/9873029/DG6HD-1BW.js.txt)

1. remove '.txt' from the filename
2. copy the file to your zigbee2mqtt server (I use WinSCP to upload to /config/zigbee2mqtt on my Home Assistant OS server)
3. in the Z2M GUI, add 'DG6HD-1BW.js' an external converter under 'Settings->External converters'
4. restart Z2M (I restart mine from the Home Assistant Add-On)

**Comment by BradNagel** [comment]
On the dev console for the device you can select what endpoints and clusters you want to read\write to. Endpoint 1 Cluster LevelCtl will have attributes onLevel, onTranstionTime and offTransitionTime. Cluster lightingBallastCfg has the powerOnLevel, Dimmer min and max. 

I later found a PDF manual showing the button combination to set min and max levels using the switch itself. I'm wondering if the switch has more ways to make these changes that are not documented.

**Reply by nebhead** [reply]
Thanks for the quick reply!  

According the response I got from Leviton Support, they only support the documented paddle programming that is listed in the manual.  The other features can only be programmed through zigbee.  

Anyway, thanks for the tips.  I'll take a look at the dev console and see if I can get it setup.  If I run into trouble, I might have some follow-up questions.  Thanks!  

Maybe we should file an issue/feature request to get this stuff added in for this device.

**Reply by nebhead** [reply]
It all seems to work!  Well, except for one thing.  When I use Home Assistant (or Z2M) to turn off the switch, then use the physical switch to turn it on again, it has reverted to turning on previous brightness level.  And for some reason, that previous brightness level is the dimmest level.  Any thoughts on how to fix this issue?

**Reply by BradNagel** [reply]
That is controlled by the onLevel attribute. For myself I set it to 254 to always have it full brightness. I think 255 is the default to use the previous brightness level. powerOnLevel I think is the same way.

Yeah, A request to add these would be awesome. I was thinking of learning how to do it myself but I just don't have the time to do it.

**Reply by nebhead** [reply]
> That is controlled by the onLevel attribute. For myself I set it to 254 to always have it full brightness. I think 255 is the default to use the previous brightness level. powerOnLevel I think is the same way.

You are a genius!  Thank you for this information.  Seems to work for me.  

> Yeah, A request to add these would be awesome. I was thinking of learning how to do it myself but I just don't have the time to do it.

Same.  I'll send up the request with the information and hope for the best.  Maybe someday I will get some time to learn how to make edits to the database.

**Reply by nebhead** [reply]
FYI:  Raised a feature request here: https://github.com/Koenkk/zigbee2mqtt/issues/13928
